homeassistant/components/homeassistant_hardware/firmware_config_flow.py

```diff
@@ -449,22 +408,12 @@ async def async_step_prepare_thread_installation(
         otbr_manager = get_otbr_addon_manager(self.hass)
         addon_info = await self._async_get_addon_info(otbr_manager)
 
-        if addon_info.state == AddonState.RUNNING:
-            # Stop the addon before continuing to flash firmware
-            await otbr_manager.async_stop_addon()
-
-        return await self.async_step_install_thread_firmware()
-
-    async def async_step_finish_thread_installation(
-        self, user_input: dict[str, Any] | None = None
-    ) -> ConfigFlowResult:
-        """Finish Thread installation by starting the OTBR addon."""
-        otbr_manager = get_otbr_addon_manager(self.hass)
-        addon_info = await self._async_get_addon_info(otbr_manager)
-
         if addon_info.state == AddonState.NOT_INSTALLED:
             return await self.async_step_install_otbr_addon()
 
+        if addon_info.state == AddonState.RUNNING:
+            await otbr_manager.async_stop_addon()
```

Comment:
Don't we need to stop the add-on before flashing?

Reply:
Ok.

We could consider restarting the add-on after the configuration instead of stopping and starting.